controllers/nodehealthcheck_controller.go

```diff
@@ -106,32 +105,69 @@ func (r *NodeHealthCheckReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		}
 	}()
 
+	// check if we need to disable NHC because of invalid configuration
+	// Remove this and corresponding test when kubebuilder supports minimum on IntOrStr types
+	if err = utils.ValidateMinHealthy(nhc); err != nil {
+		// update status if needed
+		if !utils.IsConditionTrue(nhc.Status.Conditions, remediationv1alpha1.ConditionTypeDisabled, remediationv1alpha1.ConditionReasonDisabledInvalidConfig) {
+			log.Info("disabling NHC because of invalid config")
+			meta.SetStatusCondition(&nhc.Status.Conditions, metav1.Condition{
+				Type:    remediationv1alpha1.ConditionTypeDisabled,
+				Status:  metav1.ConditionTrue,
+				Reason:  remediationv1alpha1.ConditionReasonDisabledInvalidConfig,
```

Comment:
I think it's worth using a more specific reason (i.e invalid min healthy) since in the future we might have different reasons for invalid config.
If you feel it's an overkill, maybe at least mention it in the log message.